Consul.Test/SemaphoreTest.cs

```diff
@@ -417,5 +417,54 @@ public async Task Cancelling_A_Token_When_Acquiring_A_Lock_Should_Throw_LockNotH
             await distributedLock2.Destroy();
             masterClient.Dispose();
         }
+
+        [Fact]
+        public async Task Cancelling_A_Token_When_Acquiring_A_Lock_Respects_The_Token()
+        {
+            const string key = "service/myApp/leader";
```

Comment:
Using the same key name across several tests can affect other tests using same name of the key.
If one of the tests fails then then state of the agent will not be "clean" thus other tests can also fail in unexpected way.
For better test isolation I would suggest using some unique name for the key, e.g.:
```
var keyName = Path.GetTempFileName();
```

Reply:
Done